lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Balance.hs

```diff
@@ -576,10 +576,14 @@ balanceTransaction
     s
     partialTx
     = do
-    let adjustedPartialTx = flip (over #tx) partialTx $
-            assignMinimalAdaQuantitiesToOutputsWithoutAda
-                (recentEra @era)
-                (pparamsLedger pp)
+    let adjustedPartialTx =
+            partialTx & over #tx
+                ( fromCardanoApiTx
+                . assignMinimalAdaQuantitiesToOutputsWithoutAda
+                    (recentEra @era)
+                    (pparamsLedger pp)
+                . toCardanoApiTx
+                )
```

Comment:
TODO: adjust `assignMinimalAdaQuantitiesToOutputsWithoutAda` to modify a transaction of type `Write.Tx`.